onnxruntime/core/optimizer/transformer_memcpy.cc

```diff
@@ -184,12 +184,14 @@ void TransformerMemcpyImpl::ProcessDefs(onnxruntime::Node& node, const KernelReg
             initializers_consumed[arg.Name()] = initializer_tensor_proto;
           }
 
-          // implicit inputs have no location info in the kernel def, so treat them as provider inputs.
+          // implicit inputs have no location info in the kernel def, so do nothing to them.
```

Comment:
Can you make it clearer, such as relying on the control flow op (loop, scan, if) to do the necessary copy if the input crosses different provider.